server/src/jira/token.guard.ts

```diff
@@ -0,0 +1,14 @@
+import { Injectable, ExecutionContext, CanActivate, UnauthorizedException } from "@nestjs/common";
+import { Request } from "express";
+import { SessionData } from "express-session";
+
+@Injectable()
+export class SessionTokenGuard implements CanActivate {
+  async canActivate(context: ExecutionContext) {
+    const req: Request = context.switchToHttp().getRequest();
+    const session: SessionData = req.session;
+    if (!session.user || !session.user.refreshToken || !session.user.accessToken)
+      throw new UnauthorizedException();
+    return true;
+  }
+}
```

Comment:
Here's how I would rewrite this guard and why:

- There's no need to throw an `UnauthorizedException` from guards as Nest does it automatically if `canActivate` does not return exactly `true`.
- I kind of turned the logic around so that the `canActivate` implementation returns `false` by default (because denying access is the safest option) and returning `true` becomes a special case which happens only if the `if` clause conditions pass.

```typescript
import { CanActivate, ExecutionContext, Injectable } from "@nestjs/common";
import { Request } from "express";

@Injectable()
export class SessionTokenGuard implements CanActivate {
  async canActivate(context: ExecutionContext) {
    const request = context.switchToHttp().getRequest<Request>();
    const session = request.session;

    if (session.user?.refreshToken && session.user.accessToken) {
      return true;
    }

    return false;
  }
}
```